include/boost/geometry/formulas/gnomonic_intersection.hpp

```diff
@@ -0,0 +1,147 @@
+// Boost.Geometry
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_GNOMONIC_INTERSECTION_HPP
+#define BOOST_GEOMETRY_FORMULAS_GNOMONIC_INTERSECTION_HPP
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/cs.hpp>
+
+#include <boost/geometry/arithmetic/cross_product.hpp>
+#include <boost/geometry/formulas/gnomonic_spheroid.hpp>
+#include <boost/geometry/geometries/point.hpp>
+#include <boost/geometry/util/math.hpp>
+
+
+namespace boost { namespace geometry { namespace formula
+{
+
+/*!
+\brief The intersection of two geodesics using spheroidal gnomonic projection
+       as proposed by Karney.
+\author See
+    - Charles F.F Karney, Algorithms for geodesics, 2011
+      https://arxiv.org/pdf/1109.4448.pdf
+    - GeographicLib forum thread: Intersection between two geodesic lines
+      https://sourceforge.net/p/geographiclib/discussion/1026621/thread/21aaff9f/
```

Comment:
It is MIT-licensed, also open and we change license here to Boost. It is reworked to templates and Karney suggested this.  I don't know how to exactly formulate that here